content/4_Gold/String_Hashing.mdx

```diff
@@ -214,26 +230,56 @@ that). However, we need to count the **distinct** permutations of $N$ in $H$, so
 we need to be a bit more clever.
 
 One way to solve this is by storing the polynomial hashes of each match in a
-hashset, since we expect different permutations to have different polynomial
-hashes. The answer would simply be the size of that hashset at the end.
-
-Since the test data for this particular problem is very strong, we will probably
-get hash collisions with only one hash. To remedy this, we use two hashes for
-each match - this significantly decreases the probability of collisions.
+set, since we expect different permutations to have different polynomial
+hashes. The answer would simply be the size of that set at the end.
 
-Using the base $9973$ with the two modulos $10^9 + 9$ and $10^9 + 7$ works for
-this problem. (Note that using two bases with the same modulo works too.)
+Using a relatively small modulus such as $M=10^9+9$ will not pass (see the 
+note above regarding the birthday paradox). Instead, we use $M=2^{61}-1$.
 
 <LanguageSection>
 
 <CPPSection>
 
 ```cpp
 #include <bits/stdc++.h>
-typedef long long ll;
 using namespace std;
 
-const ll P = 9973, M1 = 1e9 + 9, M2 = 1e9 + 7;
+using ll = long long;
+//BeginCodeSnip{HashedString}
+class HashedString {
+  private:
```

Comment:
these are spaces?

Reply:
that's intended

Reply:
well, that's just how my auto-formatter formats it